From 9a5c51a3c56c089890832560e27cf1d46795c255 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Tue, 28 Jun 2022 12:18:32 -0700
Subject: [PATCH] Define WIN32_LEAN_AND_MEAN before ws2tcpip.h

Including `ws2tcpip.h` ends up including `wincrypt.h` which defines symbols which collide with OpenSSL structures, such as `X509_NAME`, unless `WIN32_LEAN_AND_MEAN` is present.
---
 crypto/includes/ngtcp2/ngtcp2_crypto.h | 3 +++
 lib/includes/ngtcp2/ngtcp2.h           | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/crypto/includes/ngtcp2/ngtcp2_crypto.h b/crypto/includes/ngtcp2/ngtcp2_crypto.h
index d0df5dfe..7ddba40d 100644
--- a/crypto/includes/ngtcp2/ngtcp2_crypto.h
+++ b/crypto/includes/ngtcp2/ngtcp2_crypto.h
@@ -32,6 +32,9 @@ extern "C" {
 #endif
 
 #ifdef WIN32
+#  ifndef WIN32_LEAN_AND_MEAN
+#    define WIN32_LEAN_AND_MEAN
+#  endif
 #  include <ws2tcpip.h>
 #endif /* WIN32 */
 
diff --git a/lib/includes/ngtcp2/ngtcp2.h b/lib/includes/ngtcp2/ngtcp2.h
index a896decf..d29a45fd 100644
--- a/lib/includes/ngtcp2/ngtcp2.h
+++ b/lib/includes/ngtcp2/ngtcp2.h
@@ -52,6 +52,9 @@
 
 #ifndef NGTCP2_USE_GENERIC_SOCKADDR
 #  ifdef WIN32
+#    ifndef WIN32_LEAN_AND_MEAN
+#      define WIN32_LEAN_AND_MEAN
+#    endif
 #    include <ws2tcpip.h>
 #  else
 #    include <sys/socket.h>
-- 
2.37.0.windows.1

